hs_core/tasks.py

```diff
@@ -477,6 +474,63 @@ def create_bag_by_irods(resource_id):
         raise ObjectDoesNotExist('Resource {} does not exist.'.format(resource_id))
 
 
+@shared_task
+def delete_resource_task(resource_id, request_username=None):
```

Comment:
wow, you've been busy!  You've implemented more tasks than I realized.

Reply:
We really need to figure out something for the file manipulation and an atomic transaction.  In my test instance, I kicked off a delete and then revoked the task.  All the db data is still intact but the resource files have been deleted.  I'll give this some thought.

Perhaps we should disable the abort task option for delete until we figure something out?

Reply:
There is https://docs.djangoproject.com/en/3.0/topics/db/transactions/#django.db.transaction.on_commit 

But it'd be nice to have a hook at the model level, I'm still looking around.

Reply:
Agreed. I'll look around as well to see if I can figure this out also before we go with the work around to disable the abort task option.

Reply:
@sblack-usu @alvacouch I took a look at this further, and believe files are deleted when ResourceFile metadata objects are deleted as part of cascade delete. I don't think on_commit would help with cascade delete on unless we make iRODS file delete to be soft delete, so we can restore files when delete action is aborted. I'd suggest to address this as a separate issue. To alleviate this situation, I'll remove abort task operation for delete action so that users cannot abort any delete tasks for now, which should prevent majority (if not all) of these partial delete issues we have run into.

Reply:
https://docs.djangoproject.com/en/1.11/topics/db/transactions/#django.db.transaction.on_commit
```
The function you pass in will be called immediately after a hypothetical database write made where on_commit() is called would be successfully committed.

If you call on_commit() while there isn’t an active transaction, the callback will be executed immediately.

If that hypothetical database write is instead rolled back (typically when an unhandled exception is raised in an atomic() block), your function will be discarded and never called.
```

Looks like we can figure something out, but I'm fine with just removing the abort task for now.